src/data/list/basic.lean

```diff
@@ -1304,6 +1305,10 @@ end insert_nth
 
 @[simp] lemma map_nil (f : α → β) : map f [] = [] := rfl
 
+theorem map_eq_foldr (f : α → β) (l : list α)
+  : map f l = foldr (λ a bs, f a :: bs) [] l :=
```

Comment:
Might as well use dot notation when possible? Also below.

Reply:
I dislike dot notation and therefore generally avoid it unless it allows me to omit a namespace. But I can also dotify everything if you find this clearer.